src/Symfony/Component/Security/Core/User/ChainUserProvider.php

```diff
@@ -104,4 +104,20 @@ public function supportsClass($class)
 
         return false;
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function upgradePassword(UserInterface $user, string $newEncodedPassword): void
+    {
+        foreach ($this->providers as $provider) {
+            if ($provider instanceof PasswordUpgraderInterface) {
+                try {
+                    $provider->upgradePassword($user, $newEncodedPassword);
+                } catch (UnsupportedUserException $e) {
```

Comment:
Wouldn't it make sense to make the call above depending on the `supportsClass()` method of the chained provider instead of catching the exception?

Reply:
That's how `refreshUser()` does it, and this should be pretty rare actually.